src/main/java/com/mariamkatamashvlii/gym/service/implementation/TrainingServiceImpl.java

```diff
@@ -0,0 +1,57 @@
+package com.mariamkatamashvlii.gym.service.implementation;
+
+import com.mariamkatamashvlii.gym.dto.trainingDto.TrainingRequest;
+import com.mariamkatamashvlii.gym.entity.Trainee;
+import com.mariamkatamashvlii.gym.entity.Trainer;
+import com.mariamkatamashvlii.gym.entity.Training;
+import com.mariamkatamashvlii.gym.exception.TraineeNotFoundException;
+import com.mariamkatamashvlii.gym.exception.TrainerNotFoundException;
+import com.mariamkatamashvlii.gym.repository.TraineeRepository;
+import com.mariamkatamashvlii.gym.repository.TrainerRepository;
+import com.mariamkatamashvlii.gym.repository.TrainingRepository;
+import com.mariamkatamashvlii.gym.service.TrainingService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.UUID;
+
+@Slf4j
+@RequiredArgsConstructor
+@Service
+public class TrainingServiceImpl implements TrainingService {
+    private final TrainingRepository trainingRepo;
+    private final TraineeRepository traineeRepo;
+    private final TrainerRepository trainerRepo;
+
+    @Override
+    public void create(TrainingRequest trainingRequest) {
+        String transactionId = UUID.randomUUID().toString();
```

Comment:
Why do you need this UUID?
And, in addition, operations od data create/update should be wrapped with transaction. Looks like it's missed here. That relates to all the services
I am about @Transactional